api-registration/server/index.js

```diff
@@ -0,0 +1,75 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+
+  try {
+    const hashedPassword = argon2.hash(password);
```

Comment:
`argon2.hash` returns a promise so here `hashedPassword` wont be the actual `hashedPassword`. You need to use a `.then` to get the actual `hashedPassword`:
```JAVASCRIPT
argon2.hash(password)
  .then(hashedPassword => {
    // Do SQL query here
    // THEN
    // Send response
  });
```